extensions/wikidata/src/org/openrefine/wikidata/qa/ConstraintFetcher.java

```diff
@@ -134,4 +135,27 @@
      * Can this property be used on items?
      */
     boolean usableOnItems(PropertyIdValue pid);
+
+    /**
+     * Returns the list of PropertyIdValues of conflicting statements
+     *
+     * @param pid:
+     *            the property having conflicts-with constraint
+     * @return
+     */
+    List<Value> getConflictsWithProperties (PropertyIdValue pid);
+
+    /**
+     * Returns the list of all the conflicting values
+     *
+     * @param pid:
+     *            the property having conflicts-with constraint
+     * @return
+     */
+    List<Value> getItemwithConflicts (PropertyIdValue pid);
```

Comment:
Didn't we agree that we needed two parameters here: the property on which the constraint is defined, and the property it conflicts with?

Reply:
That's what I want to discuss with you, that If we are following wikibase quality constraints implementation so I guess they have used the same kind of implementation, they are fetching the item list at once and then checking that if the conflicting property is having a value from that list. https://github.com/wikimedia/mediawiki-extensions-WikibaseQualityConstraints/blob/master/src/ConstraintCheck/Checker/ConflictsWithChecker.php#L119-L139

Reply:
Their architecture is quite different from what we have here: yes they fetch items in a single go, but in this context they are already looking at a single constraint definition anyway. We should make sure examples like https://www.wikidata.org/wiki/Property:P4013 are handled correctly.

Reply:
Happy to discuss that at our next meeting on tuesday if that helps :)

Reply:
@wetneb: I have made the changes proposed, still finding an issue. When there are multiple issues to be reported for conflicts with constraint only the one is reported at once. I have checked that the raiseWarning method is returning true for all the possible issues but still the issues are not flagged in the issues tab. Please help. Thank you.